src/transformers/benchmark/benchmark_utils.py

```diff
@@ -339,3 +375,177 @@ def bytes_to_human_readable(memory_amount: int) -> str:
             return "{:.3f}{}".format(memory_amount, unit)
         memory_amount /= 1024.0
     return "{:.3f}TB".format(memory_amount)
+
+
+class Benchmarks(ABC):
+    """
+    Benchmarks is a simple but feature-complete benchmarking script
+    to compare memory and time performance of models in Transformers.
+    """
+
+    args: BenchmarkArguments
+    configs: PretrainedConfig
+    print_fn: Callable[[str], None]
+
+    def __init__(self, args: BenchmarkArguments = None, configs: PretrainedConfig = None):
+        self.args = args
+        self.print_fn = self.get_print_function(args)
+
+        if configs is None:
```

Comment:
The idea is here the following: 
We want to allow both measuring time / memory of our "standard" models, but also of specific models given just their config. If the config is None => we use the standard models according to the `model_names` else we init the model class with a specific config.